src/Microsoft.DotNet.Build.Tasks/VersionTools/UpdateDependencies.cs

```diff
@@ -42,96 +37,92 @@ public class UpdateDependencies : Task
-            {
-                Trace.Listeners.RemoveMsBuildTraceListeners(listeners);
-            }
-
-            return true;
-        }
+                    Regex upgrader = CreateXmlUpdateRegex($"{info.Name}{s_currentRef}", s_currentRef);
 
-        private IEnumerable<IDependencyUpdater> GetDependencyUpdaters()
-        {
-            if (ProjectJsonFiles?.Any() ?? false)
-            {
-                yield return new ProjectJsonUpdater(ProjectJsonFiles.Select(item => item.ItemSpec));
+                    Action replace = FileUtils.ReplaceFileContents(
+                        CurrentRefXmlPath,
+                        contents =>
+                        {
+                            Match m = upgrader.Match(contents);
+                            Group g = m.Groups[s_currentRef];
+
+                            return contents
+                                .Remove(g.Index, g.Length)
+                                .Insert(g.Index, masterSha);
+                        });
+                    replace();
+                }
             }
 
-            if (XmlUpdateStep != null)
+            if (updateResults.ChangesDetected())
             {
-                foreach (ITaskItem step in XmlUpdateStep)
-                {
-                    string buildInfoName = step.GetMetadata("BuildInfoName");
-                    string packageId = step.GetMetadata("PackageId");
-
-                    FileRegexUpdater updater;
+                var origin = new GitHubProject(ProjectRepoName, GitHubUser);
 
-                    if (!string.IsNullOrEmpty(buildInfoName))
-                    {
-                        updater = new FileRegexReleaseUpdater
-                        {
-                            BuildInfoName = buildInfoName
-                        };
-                    }
-                    else
-                    {
-                        updater = new FileRegexPackageUpdater
-                        {
-                            PackageId = packageId
-                        };
-                    }
-                    updater.Path = step.GetMetadata("Path");
-                    updater.Regex = new Regex($@"<{step.GetMetadata("ElementName")}>(?<version>.*)<");
-                    updater.VersionGroupName = "version";
+                var upstreamBranch = new GitHubBranch(
+                    ProjectRepoBranch,
+                    new GitHubProject(ProjectRepoName, ProjectRepoOwner));
 
-                    yield return updater;
+                string suggestedMessage = updateResults.GetSuggestedCommitMessage();
+                string body = string.Empty;
+                if (NotifyGitHubUsers != null)
+                {
+                    body += PullRequestCreator.NotificationString(NotifyGitHubUsers.Select(item => item.ItemSpec));
                 }
+
+                var prCreator = new PullRequestCreator(gitHubAuth, origin, upstreamBranch, GitHubAuthor);
+                prCreator.CreateOrUpdateAsync(
```

Comment:
Looking further it looks like that would be VerifyDependencies -FixDependencies? If so it might be worth updating the name some so that it is clear which does which. I would have expected to call UpdateDependencies.